### PR TITLE
Add new eventids for work item success metrics

### PR DESCRIPTION
**Pull request by rtaket**
Note that the existing EventIds have changed.

Here's a query that contains most of the new EventIds. It does not contain 9002 because there's currently no transformer that can return null.

traces
| where operation_Id == "ebcc7f10-5525-415f-affa-51827c373f98"
| order by timestamp asc

### CODE REVIEW

File: src/Sarif.WorkItems/SarifWorkItemFiler.cs [hunk 1]
```diff
@@ -46,6 +47,15 @@ public SarifWorkItemFiler(Uri filingUri = null, SarifWorkItemContext filingConte
             this.FilingClient = FilingClientFactory.Create(this.FilingContext.HostUri);
 
             this.Logger = ServiceProviderFactory.ServiceProvider.GetService<ILogger<FilingClient>>();
+
+            Assembly currentAssembly = Assembly.GetExecutingAssembly();
```
Comment (michaelcfanning) [inline src/Sarif.WorkItems/SarifWorkItemFiler.cs:51]: >currentAssembly [](start = 21, length = 15)

Throw this in a helper somewhere? Wouldn't this be useful to invoke for all the transformers as well? I think I suggest putting an extension method on the Assembly type. So you can do:

Assembly.GetExecutingAssembly().LogIdentity() or some such.

File: src/Sarif.WorkItems/SarifWorkItemFiler.cs [hunk 2]
```diff
@@ -224,13 +237,18 @@ private void FileWorkItemsHelper(SarifLog sarifLog, SarifWorkItemContext filingC
                 //
                 UpdateLogWithWorkItemDetails(sarifLog, sarifWorkItemModel.HtmlUri, sarifWorkItemModel.Uri);
 
-                LogMetricsForFiledWorkItem(sarifLog, sarifWorkItemModel, FilingResult.Succeeded);
+                LogMetricsForProcessedModel(sarifLog, sarifWorkItemModel, FilingResult.Succeeded, null);
```
Comment (michaelcfanning) [inline src/Sarif.WorkItems/SarifWorkItemFiler.cs:240]: >null [](start = 98, length = 4)

provide parameter name here for clarity (or make the underlying argument optional)

File: src/Sarif.WorkItems/SarifWorkItemFiler.cs [hunk 3]
```diff
@@ -224,13 +237,18 @@ private void FileWorkItemsHelper(SarifLog sarifLog, SarifWorkItemContext filingC
...
             }
             catch (Exception ex)
             {
                 this.Logger.LogError(ex, "An exception was raised filing log '{logGuid}'.", logGuid);
                 Console.Error.WriteLine(ex);
```
Comment (michaelcfanning) [inline src/Sarif.WorkItems/SarifWorkItemFiler.cs:245]: >ole.Error.WriteLine(ex); [](start = 20, length = 24)

delete this console write?

File: src/Sarif.WorkItems/SarifWorkItemFiler.cs [hunk 4]
```diff
@@ -224,13 +237,18 @@ private void FileWorkItemsHelper(SarifLog sarifLog, SarifWorkItemContext filingC
...
-                LogMetricsForFiledWorkItem(sarifLog, sarifWorkItemModel, FilingResult.ExceptionRaised);
+
+                Dictionary<string, object> customDimentions = new Dictionary<string, object>();
```
Comment (michaelcfanning) [inline src/Sarif.WorkItems/SarifWorkItemFiler.cs:247]: >string [](start = 27, length = 6)

You should put this whole catch block in a helper that accepts the exception and the model.

File: src/Sarif.WorkItems/SarifWorkItemModel.cs [hunk 5]
```diff
@@ -16,15 +16,19 @@ public class SarifWorkItemModel : WorkItemModel<SarifWorkItemContext>
 
         public IList<Uri> LocationUris { get; private set; }
 
+        public Guid Id { get; }
```
Comment (michaelcfanning) [inline src/Sarif.WorkItems/SarifWorkItemModel.cs:19]: >Id [](start = 20, length = 2)

Let's name this guid.

File: src/Sarif.WorkItems/SarifWorkItemFiler.cs [hunk 6]
```diff
@@ -268,24 +287,43 @@ private void LogMetricsForFiledWorkItem(SarifLog sarifLog, SarifWorkItemModel sa
...
-                    { "labelsOrTags", tags },
-                    { "locationUri", uris },
-                    { "milestone", sarifWorkItemModel.Milestone },
-                    { "ownerOrAccount", sarifWorkItemModel.OwnerOrAccount },
-                    { "repositoryOrProject", sarifWorkItemModel.RepositoryOrProject },
-                    { "title", sarifWorkItemModel.Title },
-                    { "uri", sarifWorkItemModel.Uri },
+                    { "LogGuid", logGuid },
+                    { "WorkItemModelId", sarifWorkItemModel.Id },
+                    { "Area", sarifWorkItemModel.Area },
```
Comment (michaelcfanning) [inline src/Sarif.WorkItems/SarifWorkItemFiler.cs:292]: >Area [](start = 23, length = 4)

you can use nameof() for most of these now

File: src/Sarif.WorkItems/SarifWorkItemFiler.cs [hunk 7]
```diff
@@ -295,16 +333,16 @@ private void LogMetricsForFiledWorkItem(SarifLog sarifLog, SarifWorkItemModel sa
...
-                    { "ruleId", ruleMetrics.RuleId },
-                    { "errorCount", ruleMetrics.ErrorCount },
-                    { "warningCount", ruleMetrics.WarningCount },
-                    { "noteCount", ruleMetrics.NoteCount },
-                    { "openCount", ruleMetrics.OpenCount },
-                    { "reviewCount", ruleMetrics.ReviewCount },
-                    { "suppressedByTransformerCount", ruleMetrics.SuppressedByTransformerCount }
+                    { "LogGuid", logGuid },
+                    { "WorkItemModelId", sarifWorkItemModel.Id },
+                    { "Tool", ruleMetrics.Tool },
```
Comment (michaelcfanning) [inline src/Sarif.WorkItems/SarifWorkItemFiler.cs:338]: >Tool [](start = 23, length = 4)

ditto, nameof().

**Comment by alisonmaclellan** [comment]
Thanks Ryley!

**Pull request merged**